app/static/js/userProfile.js

```diff
@@ -230,6 +230,99 @@ $(document).ready(function(){
     });
   });
 
+   /* =========================
+   * NOTES: Edit functionality
+   * ========================= */
+
+  // Open the Edit Note modal and prefill fields
+  $("#profileTable").on("click", ".editNoteButton", function () {
+    const noteId   = $(this).data("noteid");
+    const username = $(this).data("username");
+    const content  = $(this).data("content") || "";
+    const viewTier = String($(this).data("viewtier") || "");
+    const isBonner = (($(this).data("isbonner") || "") === "yes");
+
+    // Hidden fields
+    $("#editNoteId").val(noteId);
+    $("#editNoteUsername").val(username);
+
+    // Text
+    $("#editNoteTextArea").val(content);
+
+    // Visibility (if present for this user)
+    const $vis = $("#editNoteVisibility");
+    if ($vis.length) {
+      if ($vis.find(`option[value="${viewTier}"]`).length) {
+        $vis.val(viewTier);
+      } else {
+        $vis.val("1");
+      }
+    }
+
+    // Bonner toggle (if present)
+    const $bonnerToggle = $("#editBonnerToggle");
+    if ($bonnerToggle.length) {
+      $bonnerToggle.prop("checked", isBonner);
+    }
+
+    $("#editNoteModal").modal("show");
+  });
+
+  // Save changes from the Edit Note modal
+  $("#editNoteForm").on("submit", function (e) {
```

Comment:
Considering the Add Note modal, the layout should be similar, meaning that Visibility and Bonner program toggle should be up above the note text box